File: node.js/bot.js
const TelegramBot = require('node-telegram-bot-api');
const fastify = require('fastify')();
const TOKEN = 'YOUR-BOT-API'; //enter your bot API 

const bot = new TelegramBot(TOKEN, { polling: true });

const channelId = '@ID'; //Enter your channel ID

async function checkTimeLoop() {
  while (true) {
    const now = new Date();
    let hours = now.getHours();
    let minutes = now.getMinutes();

if (hours === minutes) {
  bot.sendMessage(channelId, `${hours}:${minutes}`).then(() => {
    console.log(` massage sent : ${hours}:${minutes}`);
  }).catch((error) => {
    console.error('there was an ERROR ! : ', error);
  });
} else {
    console.log('');
}
    await delay(50000); 
  }
}

function delay(ms) {
  return new Promise(resolve => setTimeout(resolve, ms));
}

checkTimeLoop();

bot.on('message', (msg) => {
  const chatId = msg.chat.id;
  const text = msg.text;
  const name = msg.username;

  bot.sendMessage(chatId, 'do not Text me !');
});

fastify.get('/', function (request, reply) {
  reply.send({ message: 'telegram bot is activate  ' });
});

fastify.listen(process.env.PORT || 3000, (err, address) => {
  if (err) {
    console.error(err);
    process.exit(1);
  }
  console.log(`server is running ${address}`);
});

File: pyhton/bot.py
import asyncio
import telegram
from telegram.ext import Updater, CommandHandler, MessageHandler, Filters
from flask import Flask 

TOKEN = 'YOUR_TELEGRAM_BOT_API' #ENTER YOUR TELEGRAM BOT API
CHANNEL_ID = '@your_channel_id'  # ENTER YOUR CHANNEL ID

bot = telegram.Bot(token=TOKEN)
app = Flask(__name__)

async def check_time_loop():
    while True:
        now = asyncio.get_event_loop().run_in_executor(None, lambda: telegram.utils.helpers.now())
        now = await now
        hours = now.hour
        minutes = now.minute

        if hours == minutes:
            try:
                await bot.send_message(chat_id=CHANNEL_ID, text=f'{hours}:{minutes}')
                print(f"Message sent: {hours}:{minutes}")
            except Exception as e:
                print(f"Error in sending message: {e}")
        await asyncio.sleep(50)  

def handle_message(update, context):
    chat_id = update.message.chat_id
    context.bot.send_message(chat_id=chat_id, text="Do not text me!")

@app.route('/')
def home():
    return 'Telegram bot is activated'

def main():
    updater = Updater(TOKEN, use_context=True)
    dp = updater.dispatcher

    dp.add_handler(MessageHandler(Filters.text, handle_message))

    updater.start_polling()

    loop = asyncio.get_event_loop()
    loop.create_task(check_time_loop())

    app.run(port=3000)

if __name__ == '__main__':
    main()